cleanup/artifactCleanup/artifactCleanup.groovy

```diff
@@ -218,3 +272,21 @@ private def getSkippedPaths(String[] repos) {
     log.info "Elapsed time to retrieve paths to skip: " + duration
     return skip
 }
+
+private def mapTimeUnitToCalendar (String timeUnit) {
+    switch ( timeUnit ) {
+        case "minute":
+            return Calendar.MINUTE
+        case "hour":
+            return Calendar.HOUR
+        case "day":
+            return Calendar.DAY_OF_YEAR
+        case "month":
+            return Calendar.MONTH
+        case "year":
+            return Calendar.YEAR
+        default:
+            log.info "$timeUnit is no valid time unit. Falling back to 'month'"
```

Comment:
We should throw an exception here, instead of assigning a default value,  when the user try to use a not supported timeUnit value. If the user has a typo when trying to use `minute`, `hour` or `day`, falling back to `month` can be dangerous by cleaning up things that the user does not want to remove. The fall back mechanism should be in place only it the value of `timeUnit` is not defined.

Reply:
@elioengcomp While I entirely agree with your premise, I don't really see how `month` would delete anything that wouldn't also be deleted by `minute`, `hour`, or `day`. `year`, on the other hand ...

Reply:
@DarthFennec you are right. My logic should be the other way around. If I'm trying to cleanup only things not downloaded in the last year, I don't want a typo to cause me to lose artifacts not downloaded in the last month.